components/raftstore/src/store/fsm/store.rs

```diff
@@ -1695,26 +1699,37 @@ impl<EK: KvEngine, ER: RaftEngine> RaftBatchSystem<EK, ER> {
             pd_worker,
             background_worker,
             cleanup_worker: Worker::new("cleanup-worker"),
+            snap_gen_worker: Worker::new("snap-gen-worker"),
             region_worker: Worker::new("region-worker"),
             purge_worker,
             raftlog_fetch_worker: Worker::new("raftlog-fetch-worker"),
             coprocessor_host: coprocessor_host.clone(),
             refresh_config_worker: LazyWorker::new("refreash-config-worker"),
         };
         mgr.init()?;
+
+        let snap_gen_runner = SnapGenRunner::new(
+            engines.kv.clone(),
+            mgr.clone(),
+            cfg.clone(),
+            self.router(),
+            Some(Arc::clone(&pd_client)),
+        );
+
         let region_runner = RegionRunner::new(
             engines.kv.clone(),
             mgr.clone(),
             cfg.clone(),
             workers.coprocessor_host.clone(),
             self.router(),
-            Some(Arc::clone(&pd_client)),
         );
-        let snap_generator_pool = region_runner.snap_generator_pool();
+        let snap_generator_pool = snap_gen_runner.snap_generator_pool();
+        let snap_gen_scheduler = workers
+            .snap_gen_worker
+            .start("snap-gen-worker", snap_gen_runner);
```

Comment:
A new thread?

Reply:
Yeah the current approach creates a new thread. I mentioned this in the PR description:
> However, one inefficiency remains: the snap gen worker mainly forwards
tasks to the snap gen thread pool, becoming a thread that doesn't do
much work. One optimization to consider is to send tasks directly to
the thread pool without using the snap gen worker as an intermediary.

The snap gen worker thread isn't strictly necessary. I'll try to see how to remove it.

Reply:
I've made some change that allows the snapshot generation work (`SnapGenContext::handle_gen`) to reuse the yatp pool of the snap gen worker. So now the PR doesn't introduce a new thread pool.